CSUB-591: Updates for docker-compose.yml

atodorov opened the pull request:
# Description of proposed changes

<describe what this PR is about and why we want it>

---
Practical tips for PR review & merge:

- [ ] All GitHub Actions report PASS
- [ ] Newly added code/functions have unit tests
  - [ ] Coverage tools report all newly added lines as covered
  - [ ] The positive scenario is exercised
  - [ ] Negative scenarios are exercised, e.g. assert on all possible errors
  - [ ] Assert on events triggered if applicable
  - [ ] Assert on changes made to storage if applicable
- [ ] Modified behavior/functions - try to make sure above test items are covered
- [ ] Integration tests are added if applicable/needed

atodorov committed:
Remove old docker-compose.yml file

it doesn't seem to be used anywhere and also looks like a build compose,
not an example of how creditcoin-node should be run

atodorov committed:
Add docker-compose.yml example with bind-mounted directory

Note: testnet chain spec already specifies boot node so no need to
hard-code it here

atodorov committed:
Make sure /creditcoin-node/data is present inside the container

atodorov committed:
Add a test job for persistently mounted docker volume

jonZlotnik commented on .github/workflows/ci.yml:
```diff
@@ -51,6 +51,21 @@ jobs:
+      - name: Sanity check for persistently mounted directory
+        run: |
+          sleep 60
+          docker exec creditcoin-validator ls -ld /creditcoin-node/data
+          docker exec creditcoin-validator ls -la /creditcoin-node/data
```
Is there a useful way to assert this output in the ci?

atodorov replied:
>  Is there a useful way to assert this output in the ci?

What do you mean exactly? I'm not sure I follow.

Meanwhile I've removed the "[Make sure /creditcoin-node/data is present inside the container](https://github.com/gluwa/creditcoin/pull/1145/commits/c1357d794cf52ced0daeb6689a0e7a51b54a7f4c)" commit and retried this CI job to make sure it will fail w/o the fix:
https://github.com/gluwa/creditcoin/actions/runs/5323529135/jobs/9641539867

jonZlotnik replied:
I guess I meant that I wanted to know if we can assert that it's actually being populated properly.

atodorov replied:
Failing CI job without the fix: https://github.com/gluwa/creditcoin/actions/runs/5323529135/jobs/9641539867 

the result is that once the container starts it will try to write to `/creditcoin-node/data` which is owned by root (when no fix applied) and the container itself will crash. Subsequently the `ls` commands will not work because there's no container to execute these commands on.

jonZlotnik commented on docker-compose.yml:
```diff
@@ -1,18 +1,27 @@
+    image: gluwa/creditcoin
+    command:
+      - --name
+      - atodorov validator
```
```suggestion
      - validator
```

jonZlotnik commented on docker-compose.yml:
```diff
@@ -1,18 +1,27 @@
+      - atodorov validator
+      - --validator
+      - --bootnodes
+      - "/dns4/testnet-bootnode.creditcoin.network/tcp/30333/p2p/12D3KooWG3eEuYxo37LvU1g6SSESu4i9TQ8FrZmJcjvdys7eA3cH"
```
I think this needs to be parameterized.
I guess the test network is a good default, but if the network is killed and restarted, or boot nodes change, etc. etc. it's _yet another place_ we'll need to remember to change infra details.

atodorov replied:
>  it's yet another place we'll need to remember to change infra details.

that also holds true if you employ parameters.

jonZlotnik replied:
Yeah, I was thinking that as I was writing it. Was hoping you'd have a better idea than me for somewhere to put the network-related values like a central config file or something.
Will CI at least fail if the boot node isn't accessible?

dlebee replied:
because of PoS you might be able to query that there is no next era planned since no staking has occurred on the network, if you use the RPC of the node being started, but you'll have to wait for it to sync?

> I forgot to mention, that the reason that's true is because if it doesn't peer with a bootnode it will act as a archive node on its own until it peers, you could do a health check call and see how many peers it has.

jonZlotnik replied:
So this could become a bogus compose file in a couple of months too.
I'll note this and we can address it later in the week.

atodorov replied:
>  Will CI at least fail if the boot node isn't accessible?

Apparently not, however this is due to the fact that the boot node is made available in the chain spec.

```diff
       - validator
       - --validator
       - --bootnodes
-      - "/dns4/testnet-bootnode.creditcoin.network/tcp/30333/p2p/12D3KooWG3eEuYxo37LvU1g6SSESu4i9TQ8FrZmJcjvdys7eA3cH"
+      - "/dns4/bogus-boot-node.example.com/tcp/30333/p2p/12D3KooWG3eEuYxo37LvU1g6SSESu4i9TQ8FrZmJcjvdys7eA3cH"
       - --chain
       - test
```

Logs:
```
creditcoin-validator | 2023-06-20 19:57:13 📦 Highest known block at #67525    
creditcoin-validator | 2023-06-20 19:57:13 〽️ Prometheus exporter started at 127.0.0.1:9615    
creditcoin-validator | 2023-06-20 19:57:13 Running JSON-RPC HTTP server: addr=127.0.0.1:9933, allowed origins=["http://localhost:*", "http://127.0.0.1:*", "https://localhost:*", "https://127.0.0.1:*", "https://polkadot.js.org"]    
creditcoin-validator | 2023-06-20 19:57:13 Running JSON-RPC WS server: addr=127.0.0.1:9944, allowed origins=["http://localhost:*", "http://127.0.0.1:*", "https://localhost:*", "https://127.0.0.1:*", "https://polkadot.js.org"]    
creditcoin-validator | 2023-06-20 19:57:16 👶 Starting BABE Authorship worker    
creditcoin-validator | 2023-06-20 19:57:16 🔍 Discovered new external address for our node: /ip4/217.18.251.214/tcp/30333/p2p/12D3KooWF2R2YLC3ogzUNvoxQY26F9BwztyLShCResvYtGZVQoBg    
creditcoin-validator | 2023-06-20 19:57:17 [67852] 💨 unstaked , outcome: Ok(())    
creditcoin-validator | 2023-06-20 19:57:17 [67872] 💨 unstaked , outcome: Ok(())    
creditcoin-validator | 2023-06-20 19:57:18 ⚙️  Syncing, target=#68837 (3 peers), best: #67921 (0x2d86…1fbc), finalized #67584 (0xa1c1…8fd5), ⬇ 44.9kiB/s ⬆ 3.1kiB/s    
```

jonZlotnik replied:
Great! so let's remove it completely!

jonZlotnik reviewed (dismissed):
Noice

atodorov committed:
ci: Remaove redundant -v argument for docker run

this only make sense if we specify --base-path, which we do in the next
section
